src/Base/Vector3D.cpp

```diff
@@ -227,6 +227,25 @@ bool Vector3<float_type>::IsEqual(const Vector3<float_type>& rclPnt, float_type
     return Distance(*this, rclPnt) <= tol;
 }
 
+template<class float_type>
+bool Vector3<float_type>::IsParallel(const Vector3<float_type>& rclPnt, float_type tol) const
+{
+    Vector3<float_type> v1 = *this;
+    Vector3<float_type> v2 = rclPnt;
+    double dot = abs(v1 * v2);
+    double mag = v1.Length() * v2.Length();
+    return (abs(dot - mag) < tol);
+}
+
+template<class float_type>
+bool Vector3<float_type>::IsNormal(const Vector3<float_type>& rclPnt, float_type tol) const
+{

```

Comment:
This code suffers from the first and second point as IsParallel()

Reply:
The method doesn't even work correctly. If v1=(1,2,3) and v2(3,0,-1) then the dot product is exactly 0.0 but the check `dot < tol` fails if tol is 0.0.

Reply:
Another weakness is if you allow a certain tolerance because then the result depends on the length of the input vectors. Assuming an angular tolerance of 1 degree is allowed then tol will be 0.01745
The use the vectors v1=(10000,20000,30000) and v2(30000,00000,-10001)
`
math.pi/2 - v1.getAngle(v2)
`
gives `-2.5354374089570086e-05` which is clearly inside the tolerance but dot will be -30000.0.